bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MessageBox.java

```diff
@@ -99,6 +106,88 @@ public MessageBox (Shell parent) {
 public MessageBox (Shell parent, int style) {
 	super (parent, checkStyle (parent, checkStyle (style)));
 	checkSubclass ();
+	/* Create the callback hook */
+	int threadId = OS.GetCurrentThreadId ();
+	Callback cbtCallback = new Callback (this, "CBTProc", 3); //$NON-NLS-1$
+	cbtHook = OS.SetWindowsHookEx (OS.WH_CBT, cbtCallback.getAddress (), 0, threadId);
```

Comment:
I would rather set hook immediately before `OS.MessageBox()`

Reply:
Also, this is another leak. I suggest to call `UnhookWindowsHookEx()` immediately after the button labels are set.

Reply:
Done.